arch/Kconfig

```diff
@@ -352,6 +353,10 @@ config ARCH_HAVE_TESTSET
 	bool
 	default n
 
+config ARCH_HAVE_TLS
```

Comment:
ARCH_HAVE_TLS to ARCH_HAVE_THREAD_LOCAL?
the supported from tls can be done in the general way, only thread_local keyword need arch or compiler specific code